pkg/liveshare/client.go

```diff
@@ -112,15 +119,49 @@ func Connect(ctx context.Context, opts Options) (*Session, error) {
 	s := &Session{
 		ssh:             ssh,
 		rpc:             rpc,
+		grpc:            grpc.New(),
 		clientName:      opts.ClientName,
 		keepAliveReason: make(chan string, 1),
 		logger:          opts.Logger,
 	}
 	go s.heartbeat(ctx, 1*time.Minute)
 
+	// Connect to the gRPC server so we can make requests anywhere we have access to the session
+	s.connectToGrpcServer(ctx, opts.SessionToken)
+
 	return s, nil
 }
 
+// Connects to the gRPC server running on the host VM
+func (s *Session) connectToGrpcServer(ctx context.Context, token string) error {
+	listen, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%d", 0))
+	if err != nil {
+		return err
```

Comment:
Wrap the error message, usually any time you have more than one error returning from a function, you'll want to add context to them.